File: Back-End/SmartAttendance.DAL/DBModels/Course.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartAttendance.DAL.DBModels
{
    public class Course
    {
        [Key]
        public int Id { get; set; }
        public string Name { get; set; }
        public bool Avilable { get; set; }
        public int? DepartmentId { get; set; }
        [ForeignKey("DepartmentId")]
        public virtual Department Department { get; set; }
        public virtual ICollection<InstructorCourse>? InstructorCourses { get; set; }

    }
}

File: Front/SmartAttendance/src/app/SmartAttendance/Model/course.ts
import { InstructorCourse } from "./instructor-course";

export class Course {
    public id: number = 0;
    public name: string = '';
    public avilable: boolean = false;
    public departmentId: number = 0;
    public instructorCourses: InstructorCourse[] = [];
}
